pact.cabal

```diff
@@ -205,8 +206,8 @@ library
                      , semigroupoids >= 5.0
                      , stm >= 2.4.4.1 && < 2.6
                      , text >= 1.2.2.1 && < 1.3
+                     , time >= 1.8
```

Comment:
Can we simply support 1.9 to avoid CPP

Reply:
I wish. It may not be possible with the current nix snapshot.

Reply:
Hmm that means one of three things:
1. Get nix to accept 1.9
2. Pin to < 1.9
3. Figure out how to regress both in tests

It's branching in our formatting code, it has to be ideally monolithic, but it not, thoroughly covered.

Reply:
There's also the option to port the formatting code from thyme -- which would mean that we'll have to maintain that code ourself. In turn we would probably get better performance and changes in time can not break pact time formatting.

The time library is very well maintained and stable, so I am not too concerned for now. But on the long term we'll probably want to own all code that formats data that makes it onto the chain.

Reply:
Pinning to 1.8 is not an option, since it would break the stack and cabal builds of chainweb.

I don't think the use of CPP is that bad here. A good thing about CPP is, that it keeps the different versions of the source syntactically separate and is very easy to remove once all builds are on 1.9.

CPP here isn't used for metaprograming, but only to keep to completely different implementations. I think the former is bad, while the latter is acceptable.

Reply:
I agree that the best solution would be if we could just move to time >= 1.9.

Reply:
If we can pin `time` in chainweb nix to 1.9, then we can go with this as the only exception will be Chainweaver/BlockExplorer right? Everything else would be running/tested on 1.9? Obviously, chainweb bootstraps are on nix for now, so can we get those builds onto 1.9 with overrides.